alloy-core/src/test/java/org/starchartlabs/alloy/test/core/PreconditionsTest.java

```diff
@@ -128,4 +239,33 @@ public void checkStateFalseSupplier() throws Exception {
         Preconditions.checkState(false, () -> "message");
     }
 
```

Comment:
We should add a test for the Supplier itself being null (Preconditions.checkState("value", t -> true, (Supplier<String>) null))

Reply:
Test added